src/core/execute.c

```diff
@@ -201,6 +202,58 @@ static const char *exec_context_tty_path(const ExecContext *context) {
         return "/dev/console";
 }
 
+static int exec_context_tty_size(const ExecContext *context, unsigned *ret_rows, unsigned *ret_cols) {
+        _cleanup_free_ char *rowskey = NULL, *rowsvalue = NULL, *colskey = NULL, *colsvalue = NULL;
+        unsigned rows = UINT_MAX, cols = UINT_MAX;
+        const char *tty;
+        int r;
+
+        assert(context);
+        assert(ret_rows);
+        assert(ret_cols);
+
+        rows = context->tty_rows;
+        cols = context->tty_cols;
+
+        tty = skip_dev_prefix(exec_context_tty_path(context));
+        if (!tty) {
+                *ret_rows = rows;
+                *ret_cols = cols;
+                return 0;
+        }
+
+        rowskey = strjoin("systemd.tty.", tty, ".rows");
+        if (!rowskey)
+                return -ENOMEM;
+
+        colskey = strjoin("systemd.tty.", tty, ".columns");
+        if (!colskey)
+                return -ENOMEM;
+
+        r = proc_cmdline_get_key_many(0,
```

Comment:
Not a matter, but I'd like to add a short comment for the argument: `/* flags = */ 0`